torch/_inductor/codegen/triton.py

```diff
@@ -1904,7 +1913,7 @@ def load(self, name: str, index: sympy.Expr):
         if advance_block_ptr:
             load_buffer.writeline(advance_block_ptr)
 
-        if not self.inside_reduction or not has_rindex:
+        if not self.inside_reduction or not indexing.has_rmask():
```

Comment:
```
        tmp0 = tl.load(in_ptr0 + (r1), rmask, eviction_policy='evict_last', other=0.0)
        tmp1 = tmp0 + 4191
        tmp2 = tmp0 < 0
        tmp3 = tl.where(tmp2, tmp1, tmp0)
        tl.device_assert(((0 <= tmp3) & (tmp3 < 4191)) | ~rmask, "index out of bounds: 0 <= tmp3 < 4191")
        tmp4 = tl.load(in_ptr1 + (tmp3 + (4191*x0)), rmask, eviction_policy='evict_last')
 ```
 
 We were incorrectly cse-ing `tmp4` because it's load did not have an rindex. I've put the condition here back to what it was prior to [this pr](https://github.com/pytorch/pytorch/pull/116079).
 
 We could also make the `has_rindex` propagate correctly from tmp0 -> tmp4. Or ban indirect indexing like [here](https://github.com/pytorch/pytorch/blob/main/torch/_inductor/codegen/triton.py#L1887-L1891).

Reply:
Is the rmask for loading in_ptr1 here get propagated from the rmask used when loading from in_ptr0?

Reply:
Yes, `TritonCSEVariable` propagates mask information to intermediate variables so that indirect indexes have all the required masks.